modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/RelayUtils.java

```diff
@@ -236,6 +237,39 @@ public static boolean isDeleteRequestWithoutPayload (MessageContext msgContext)
         return false;
     }
 
+    private static boolean isEmptyPayloadStream(MessageContext messageContext) throws AxisFault {
+
+        boolean isEmpty = false;
+        BufferedInputStream bufferedInputStream = (BufferedInputStream) messageContext
+                .getProperty(PassThroughConstants.BUFFERED_INPUT_STREAM);
+        if (bufferedInputStream != null) {
+            try {
+                bufferedInputStream.reset();
+                bufferedInputStream.mark(0);
+            } catch (Exception e) {
+                // just ignore the error
+            }
+
+        } else {
+            final Pipe pipe = (Pipe) messageContext.getProperty(PassThroughConstants.PASS_THROUGH_PIPE);
+            if (pipe != null) {
+                InputStream in = pipe.getInputStream();
+                bufferedInputStream = new BufferedInputStream(in);
+                // TODO: need to handle properly for the moment lets use around 100k
+                // buffer.
```

Comment:
shall we put one line comment here?

Reply:
This change cannot be found in the current file